web/server.go

```diff
@@ -35,10 +36,14 @@ func NewMux(config Config, context command.ExecutionContext, hook Hook) *http.Se
 	httpMux.Handle("/token", tokenHandler{
 		context: context,
 	})
-	if config.JSONIngestion {
-		httpMux.Handle("/ingest", ingestHandler{
-			metricMetadataAPI: context.MetricMetadataAPI,
-		})
+	if updateAPI, ok := context.MetricMetadataAPI.(metadata.MetricUpdateAPI); ok && config.HTTPIngestion {
```

Comment:
Maybe we should log in the case where `config.HTTPIngestion` is true but the `MetricMetadataAPI` isn't a `MetricUpdateAPI`?